File: backend/src/main/java/com/backend/backend/model/Student.java
package com.backend.backend.model;

import jakarta.persistence.*;
import java.time.LocalDate;

@Entity
@Table(name = "students")
public class Student {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String fullName;
    private String admissionNo;
    private String gender;

    private LocalDate dateOfBirth;
    private LocalDate startDate;
    private LocalDate endDate;

    @ManyToOne
    @JoinColumn(name = "classroom_id")
    private ClassRoom classRoom;

    public Student() {}

    // Getters and Setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getFullName() { return fullName; }
    public void setFullName(String fullName) { this.fullName = fullName; }

    public String getAdmissionNo() { return admissionNo; }
    public void setAdmissionNo(String admissionNo) { this.admissionNo = admissionNo; }

    public String getGender() { return gender; }
    public void setGender(String gender) { this.gender = gender; }

    public LocalDate getDateOfBirth() { return dateOfBirth; }
    public void setDateOfBirth(LocalDate dateOfBirth) { this.dateOfBirth = dateOfBirth; }

    public LocalDate getStartDate() { return startDate; }
    public void setStartDate(LocalDate startDate) { this.startDate = startDate; }

    public LocalDate getEndDate() { return endDate; }
    public void setEndDate(LocalDate endDate) { this.endDate = endDate; }

    public ClassRoom getClassRoom() { return classRoom; }
    public void setClassRoom(ClassRoom classRoom) { this.classRoom = classRoom; }
}

File: frontend/types/Student.ts
// DTO for sending to backend
export interface StudentDTO {
  id?: number;           // student ID for updates
  fullName: string;
  gender: string;
  admissionNo: string;
  dateOfBirth: string;   // format: YYYY-MM-DD
  startDate: string;     // format: YYYY-MM-DD
  endDate: string;       // format: YYYY-MM-DD
  classRoomId: number;   // send only the ID to backend
}

// Type used for displaying in frontend table
export interface Student {
  id?: number;
  fullName: string;
  gender: string;
  admissionNo: string;
  dateOfBirth: string;
  startDate: string;
  endDate: string;
  classRoom?: {
    id: number;
    name: string;
    grade: number;
  }; // full classroom object for display
}
